### DISCUSSION TITLE
How does RaspAP detect if device supports Bridged AP Mode?

### DISCUSSION THREAD

**Discussion by bachelor-emgi** [RaspAP/raspap-webgui#1377] [Q&A] [answered]
Hey, does anyone know how does RaspAP detect if wlan device supports Bridged AP Mode? I have it grayed out and I know that it is supported since I had it working when I was using Raspbian Lite (now I am using Ubuntu Server 23.04). I have tried to make RaspAP config file to force it to use Bridged AP, but on each reboot it disables the slider.

**Comment by billz** [comment, marked as answer]
Ubuntu differs from other distros in that it uses netplan to implement a network bridge. For this reason it's currently unsupported:

https://github.com/RaspAP/raspap-webgui/blob/fd201207b443eef9326c4d40f6878da674d4b6a1/templates/hostapd/advanced.php#L7

There's a netplan bridge config stub located [here](https://github.com/RaspAP/raspap-webgui/blob/master/config/raspap-bridge-br0.netplan) but this is still a work in progress.

**Reply by bachelor-emgi** [reply]
Thanks, edited and configured the netplan.yaml and works like charm. (Just had to in WiFi router edit DHCP reservation since it got new IP, but after reboot no problems at all)

**Reply by billz** [reply]
Good to know! I'll carve out some time and revisit bridge support for Ubuntu.

**Reply by bachelor-emgi** [reply]
If you are gonna automate it for future, you will have to force reboot because after sudo netplan apply, user will lose connection to ssh since original internet source will lose connection and will bridge it directly to bridge and then system restarts networking automatically and stuff happens. So for Ubuntu bridge there should be in quick installer mandatory reboot if user changes in WiFi router DHCP for device without asking so it can be executed even after sudo netplan apply. And before that warning that device will get new IP address. (I have rebooted multiple times and the config just works without any issues at all)
